src/tests/PhysicsTest.cpp

```diff
@@ -131,3 +134,62 @@ TEST_F(PhysicsManagerTest, JoinCompound) {
     }
   }
 }
+
+#ifdef ESP_BUILD_WITH_BULLET
+TEST_F(PhysicsManagerTest, BulletCompoundShapeMargins) {
+  // test that all different construction methods for a simple shape result in
+  // the same Aabb for the given margin
+  LOG(INFO) << "Starting physics test: BulletCompoundShapeMargins";
+
+  std::string objectFile = Cr::Utility::Directory::join(
+      dataDir, "test_assets/objects/transform_box.glb");
+
+  initScene(objectFile);
+
+  if (physicsManager_->getPhysicsSimulationLibrary() ==
+      PhysicsManager::PhysicsSimulationLibrary::BULLET) {
+    // test joined vs. unjoined
+    esp::assets::PhysicsObjectAttributes physicsObjectAttributes;
+    physicsObjectAttributes.setString("renderMeshHandle", objectFile);
+    physicsObjectAttributes.setDouble("margin", 0.1);
+    resourceManager_.loadObject(physicsObjectAttributes, objectFile);
+
+    // get a reference to the stored template to edit
+    esp::assets::PhysicsObjectAttributes& objectTemplate =
+        resourceManager_.getPhysicsObjectAttributes(objectFile);
+
+    // add the unjoined object
+    objectTemplate.setBool("joinCollisionMeshes", false);
+    int objectId0 = physicsManager_->addObject(objectFile, nullptr);
+
+    // add the joined object
+    objectTemplate.setBool("joinCollisionMeshes", true);
+    int objectId1 = physicsManager_->addObject(objectFile, nullptr);
+
+    esp::physics::BulletPhysicsManager* bPhysManager =
+        static_cast<esp::physics::BulletPhysicsManager*>(physicsManager_.get());
+
+    std::pair<Magnum::Vector3, Magnum::Vector3> AabbScene =
+        bPhysManager->getSceneCollisionShapeAabb();
+
+    std::pair<Magnum::Vector3, Magnum::Vector3> AabbOb0 =
+        bPhysManager->getCollisionShapeAabb(objectId0);
+    std::pair<Magnum::Vector3, Magnum::Vector3> AabbOb1 =
+        bPhysManager->getCollisionShapeAabb(objectId1);
+
+    std::pair<Magnum::Vector3, Magnum::Vector3> objectGroundTruth =
```

Comment:
Is this test testing all the places where new code was added.

Reply:
Yes, there are 3 cases currently:
1. Scenes (concave mesh collision objects)
2. Joined objects (a single convex)
3. Hierarchal objects (a compound of convex components)